### ISSUE TITLE
Support ES Private Methods in TypeScript

### ISSUE THREAD

**Issue by kitsonk** [denoland/deno#5258]
Currently we support both private fields (`#foo = "bar"`) and private methods (`#foo() {}`) in the version of v8 we use in Deno.  So the following is valid in JavaScript:

```js
class Foo {
  #foo() {
    console.log("bar");
  }
  bar() {
    this.#foo();
  }
}
```

The problem is that while private fields are allowed in TypeScript, private methods are not currently allowed.  This was because private fields and private methods were two seperate proposals and private methods took longer to reach Stage 3.

We have been working around this by setting an arrow function as the initialiser for a private field, but that is actually sub-optimal as these are not "static" as part of the classes prototype and are initialised on every construction of an instance.

This is a tracking issue for https://github.com/microsoft/TypeScript/issues/37677 which deals with ES private methods in TypeScript.

**Comment by satyarohith** [comment]
Related https://github.com/denoland/deno/pull/9960

**Comment by bartlomieju** [comment]
TypeScript 4.3 landed in #9960, so closing as resolved.
